servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -2664,6 +2664,10 @@ public final Publisher<T> skipWhile(Predicate<? super T> predicate) {
         return filter(FilterPublisher.skipWhileSupplier(predicate));
     }
 
+    public final <R> Single<R> splice(BiFunction<T, Publisher<T>, R> packer) {
```

Comment:
I'm not in love with the name `splice`. I feel like splice should be joining two things together, which perhaps in a way is happening here but this feels more like an "unpack first" or something similar... IDK, I don't have strong feelings but there's still a voice in the back of my head telling me something is off.